### PR TITLE
`zig cc`: Remove broken CUDA C/C++ support.

### PR DESCRIPTION
**Pull request by alexrp**
This has been broken for a while with apparently no one noticing:

```console
❯ zig4 cc test.cu
zig: warning: debug information option '-gdwarf32' is not supported for target 'nvptx64-nvidia-cuda' [-Wunsupported-target-opt]
error: unknown target CPU 'x86-64'
note: valid target CPU values are: sm_20, sm_21, sm_30, sm_32, sm_35, sm_37, sm_50, sm_52, sm_53, sm_60, sm_61, sm_62, sm_70, sm_72, sm_75, sm_80, sm_86, sm_87, sm_89, sm_90, sm_90a, gfx600, gfx601, gfx602, gfx700, gfx701, gfx702, gfx703, gfx704, gfx705, gfx801, gfx802, gfx803, gfx805, gfx810, gfx9-generic, gfx900, gfx902, gfx904, gfx906, gfx908, gfx909, gfx90a, gfx90c, gfx940, gfx941, gfx942, gfx10-1-generic, gfx1010, gfx1011, gfx1012, gfx1013, gfx10-3-generic, gfx1030, gfx1031, gfx1032, gfx1033, gfx1034, gfx1035, gfx1036, gfx11-generic, gfx1100, gfx1101, gfx1102, gfx1103, gfx1150, gfx1151, gfx1152, gfx12-generic, gfx1200, gfx1201, amdgcnspirv
```

It seems like it would take quite a lot of effort to fit the offload compilation model properly into `zig cc` since it needs to compile both host and device code in the same invocation, and I'm not at all convinced that it's worth the effort. It's not as if we ship any of the runtime libraries needed for this stuff, so we don't really offer anything over upstream Clang. And besides, we only advertise C/C++ compilation with `zig cc` and `zig c++`; CUDA/OpenCL/HIP seem rather out of scope. We shouldn't necessarily expose every single feature that Clang has.

I also removed all the Clang headers related to GPU offload while here, which cuts installation size by ~1,250,420 bytes.

### CODE REVIEW

**Comment by alexrp** [comment]
cc @Snektron you might have opinions on this

**Comment by andrewrk** [comment]
> no one noticing

#19442

My main objection here is that it complicates our LLVM integration. If this is merged then "updating LLVM" gains another step, and now we have to explain that when you use zig C compilation, available headers are equivalent to Clang *except* for cuda.

On the other hand, status quo, while broken, is easier to understand and is a step closer to working than if this PR is merged. So I think this represents negative progress.

In conclusion, you can merge it if you want.

**Comment by alexrp** [comment]
> #19442

Huh, not sure how I didn't come across that. But also interesting that it's been broken for this long with no one trying to fix it.

> My main objection here is that it complicates our LLVM integration. If this is merged then "updating LLVM" gains another step,

Well, in fairness, the extra step is just `git rm -r lib/include/*{cuda,hip,opencl}*` as a copy-pasteable command in the upgrade doc.

> and now we have to explain that when you use zig C compilation, available headers are equivalent to Clang _except_ for cuda.

Does this actually matter? These headers are not terribly useful because we don't ship the offload runtimes with Zig. For CUDA, I think we can't even if we wanted to. (Not sure what the situation is for OpenCL and HIP.) The main selling point of `zig cc` is cross-compilation, but that's something we don't currently provide for GPU offload.

But having said that:

> On the other hand, status quo, while broken, is easier to understand and is a step closer to working than if this PR is merged. So I think this represents negative progress.

If you think that this support is in scope for the Zig project, then I'm fine with leaving it in for someone to fix if they're motivated enough. I just thought it might be nice to shed some weight here and rein in the scope of the project a bit (consider also that we aren't exposing LLVM's DXIL and SPIR-V backends).
